cli/src/main/java/com/walmartlabs/concord/cli/runner/secrets/CliSecretService.java

```diff
@@ -0,0 +1,165 @@
+package com.walmartlabs.concord.cli.runner.secrets;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2020 Walmart Inc.
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.cli.CliConfig;
+import com.walmartlabs.concord.cli.Verbosity;
+import com.walmartlabs.concord.cli.runner.VaultProvider;
+import com.walmartlabs.concord.runtime.v2.sdk.SecretService;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+import static org.fusesource.jansi.Ansi.ansi;
+
+public class CliSecretService implements SecretService {
+
+    private final List<SecretsProviderRef> secretsProviders;
+    private final VaultProvider vaultProvider;
+    private final Verbosity verbosity;
+
+    public static CliSecretService create(CliConfig cliConfig, Path workDir, Verbosity verbosity) {
+        var providers = new ArrayList<SecretsProviderRef>();
+
+        var local = cliConfig.secrets().local();
+        if (local.enabled()) {
+            var provider = new FileSecretsProvider(workDir, local.dir());
+            providers.add(new SecretsProviderRef("localFile", provider, local.writable()));
+        }
+
+        var remote = cliConfig.secrets().remote();
+        if (remote.enabled()) {
+            var provider = new RemoteSecretsProvider(workDir, remote.baseUrl(), remote.apiKey());
+            providers.add(new SecretsProviderRef("remote", provider, remote.writable()));
+        }
```

Comment:
I'm a littttle bit concerned this opens a door to a big potential oopsie. One should be very considerate of their handling of secrets from "real" environments to start with, but this could easily end up with a "I got out of a pickle with the CLI pointed at prod real quick on Monday and forgot to turn off remote secrets...Tuesday got _interesting_  😬"

That could be addressed with an extra setting for the remote config(s), or any config since the same situation _is possible_--just fewer hoops to jump through (e.g. creating specific secrets locally for `secrets.local` vs `secrets.remote` unlocking any available secret in the system). Something like `secrets.remote.warnOnAccess: true` that gives a prompt/warning that it's about to access a remote secret.

Reply:
That is a valid point. I will see if we can add a prompt there.
Or maybe providers should be configured for specific "contexts" and the CLI can require something like `--context prod" to use a non-default context?

Reply:
@benbroadaway how about now -- see the example in the PR description.
Users will need to explicitly say `concord run --context myNonDefaultContext`.

Reply:
Added the confirmation prompt as well

![image](https://github.com/user-attachments/assets/69de5bb0-64b7-44be-bda3-6b9c127b306d)